agent/xds/resources_test.go

```diff
@@ -565,6 +567,15 @@ func getAPIGatewayGoldenTestCases(t *testing.T) []goldenTestCase {
 										Remove: []string{"X-Header-Remove"},
 									},
 								},
+								RetryFilter: &structs.RetryFilter{
+									NumRetries:            pointer.Uint32(3),
+									RetryOnStatusCodes:    []uint32{500},
+									RetryOnConnectFailure: pointer.Bool(true),
+								},
```

Comment:
It looks like we don't set `RetryOn` here, but it winds up populated for us in the golden file anyway to account for status codes and connect failures. I'm wondering if we actually expect anyone to set an explicit value for `RetryOn` or if it should just always be implicitly set based on whether you include `RetryOnStatusCodes` and/or `RetryOnConnectFailure`. Thoughts?

Reply:
I think I just wasn't sure what the allowed values were and forgot to add it in. I'll add that in now.

Reply:
It's interesting that they were being set anyways 🤔

Reply:
Ahhh I found the relevant code. The gist is that Consul will add `connect-failure` and `retriable-status-codes` to `RetryOn` if the related fields are populated, so adding them ourselves in our test is redundant and will result in
```go
policy.RetryOn = []string{"connect-failure", "retriable-status-codes", "connect-failure", "retriable-status-codes"}
```

https://github.com/hashicorp/consul/blob/284e3bdb547915d82bfc89296ccad485befdee59/agent/xds/routes.go#L751C1-L786C2